src/core/utils/slaac_address.cpp

```diff
@@ -234,7 +236,7 @@ void Slaac::Update(UpdateMode aMode)
                     slaacAddr->Clear();
                     memcpy(&slaacAddr->mAddress, &prefix.mPrefix, BitVectorBytes(prefix.mLength));
 
-                    slaacAddr->mPrefixLength  = prefix.mLength;
+                    slaacAddr->mPrefixLength  = config.mOnMesh ? prefix.mLength : 128;
```

Comment:
The code and comment seems to mean that OT only generate SLAAC addresses for `on-mesh` prefixes. But the actual runtime behavior is that OT also do SLAAC for non `on-mesh` prefixes.

Reply:
I think there are two definitions for on-mesh:

1. A prefix that is assigned to the Thread network.

1. A prefix that requires using ADDR_QRY.qry when forwarding.

Both are proper uses of on-mesh. I don't have a good term that disambiguates them.

Reply:
I think we should follow the same definition of `on-mesh` as Thread Spec., in which quote `A prefix is "on-mesh" if packets with on-mesh destinations are routed within the Thread subnet`. 
So 2) is more correct according to this definition? 

Currently, `GetNextOnMeshPrefix` visits both `on-mesh` prefixes and `off-mesh` prefixes. I think we can simply rename it to `GetNextPrefix` to reduce confusion?

Reply:
If the destination is outside the Thread network, the destination does not match any prefix assigned to the Thread network. As a result, I don't consider any prefix assigned to the Thread network as "off-mesh".

This is equivalent to traditional IPv6. A prefix may be assigned to a link. Separately, a prefix may be declared as "on-link", which indicates to use IPv6 ND to discover the destination. A prefix not declared as "on-link" just means "don't use IPv6 ND to discover the destination". However, packets destined to such prefixes do not leave the link (since the prefix is assigned to the link).